src/main/java/edu/wisc/my/messages/model/Message.java

```diff
@@ -291,6 +315,55 @@ public ActionButton getConfirmButton() {
   public void setConfirmButton(ActionButton confirmButton) {
     this.confirmButton = confirmButton;
   }
+  
+  public boolean isValidToday(){
+
+   Date today = new Date();
+   
+   // Placeholder date format string, 
+   // can be overridden with an application property.
+   String formatString = "yyyy-mm-dd";
+   
+   if(env == null){
```

Comment:
Couple of things here:
1. `env` will be autowired by Spring, if `env` is null, you've got larger problems
2. I'm not sure if it's still the case, but usually, checking `env` is expensive. The default date format will never change during the runtime, so it'd be much more efficient to get the date format at static initialization (or at the very least instance instantiation) and then just reference that object from within this method.